src/helpers/clearUrl.ts

```diff
@@ -0,0 +1,44 @@
+import loadClearUrlDefinitions from "./loadClearUrlDefinitions";
+import { parseURL, serializeURL } from 'whatwg-url'
+import querystring from 'querystring'
+
+const definitions = loadClearUrlDefinitions()
+
+export default function clearUrl(url : string) : string{
+    let parsed = parseURL(url)
+    if(!parsed) return url
+
+    let queries = parsed.query ? querystring.parse(parsed.query) : {}
+
+    for(let provider of definitions){
+        if(provider.urlPattern.test(url)){
+            if(!provider.rules) continue
+            console.log('provider rules', provider.rules)
+
+            // TODO algorithm might be improvable?
+            for(let rule of provider.rules){
+                let exp = new RegExp(rule, 'g')
```

Comment:
Well we are surely not writing performance-critical codes, but judging from the code it seems that `definitions` (which comes from `loadClearUrlDefinitions`) is not changed at all? So is it possible to build all RegExps beforehand and just iterate through them on line 19? Because compiling regular expressions is known to be a relatively costly process.

Reply:
Good catch